packages/@sanity/desk-tool/src/lib/useShallowUnique.ts

```diff
@@ -0,0 +1,12 @@
+import {useRef} from 'react'
+import shallowEquals from 'shallow-equals'
+
+export function useShallowUnique<ValueType>(value: ValueType): ValueType {
```

Comment:
would be great with a comment describing the purpose and an outline of what use case(s) this solves